core/frontend/src/Viewport.ts

```diff
@@ -1126,6 +1143,11 @@ export abstract class Viewport implements IDisposable {
 
     if (this._view)
       this._view.detachFromViewport();
+
+    if (this._mapTiledGraphicsProvider) {

```

Comment:
Shouldn't this happen in Viewport.detachFromDisplayStyle, not detachFromView? And then Viewport.registerDisplayStyleListeners would create a new MapTiledGraphicsProvider if we are changing the display style?

Reply:
Fixed.